File: 2090 - K Radius Subarray Averages.py
class Solution:
    def getAverages(self, nums: List[int], k: int) -> List[int]:
        i = 0
        window = 0
        result = []
        while i < len(nums):
            if i - k < 0 or i + k >= len(nums):
                result.append(-1)
            else:
                if window == 0:
                    for j in range(i - k, i + k + 1):
                        window += nums[j]
                else:
                    window -= nums[i - k - 1]
                    window += nums[i + k]
                result.append(window // (2 * k + 1))
            i += 1
        return result

File: 2090 - K Radius Subarray Averages.cpp
class Solution {
public:
    vector<int> getAverages(vector<int>& nums, int k) {
        long int window = 0;
        vector<int> result;
        int i = 0;
        while(i < nums.size()){
            if(i - k < 0 || i + k >= nums.size()){
                result.push_back(-1);
            }else{
                if(window == 0){
                    for(int j = i - k; j < i + k + 1; j++){
                        window += nums[j];
                    }
                }else{
                    window -= nums[i - k - 1];
                    window += nums[i + k];
                }
                result.push_back(window / (2 * k + 1));
            }
            i++;
        }
        return result;
    }
};